PerfTools/JeProf/plugins/JeProfService.cc

```diff
@@ -0,0 +1,247 @@
+
+//
+//  Description: FWK service to implement hook for jemalloc heap profile
+//               dump functionality
+//
+
+#include "PerfTools/JeProf/plugins/JeProfService.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/ServiceMaker.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/FileBlock.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Run.h"
+#include <string>
+#include <dlfcn.h>
+#include <cstdio>
+#include <cstring>
+
+extern "C" {
+typedef int (*mallctl_t)(const char *name, void *oldp, size_t *oldlenp, void *newp, size_t newlen);
+}
+
+namespace {
+  bool initialize_prof();
+
+  mallctl_t mallctl = nullptr;
+  const bool have_jemalloc_and_prof = initialize_prof();
```

Comment:
Could these become member variables of the `JeProfService`?

Reply:
Just copying the pattern the was used here
https://github.com/cms-sw/cmssw/blob/master/HLTrigger/Timer/src/memory_usage.cc